Mathlib/LinearAlgebra/DirectSum/PiTensorProduct.lean

```diff
@@ -0,0 +1,70 @@
+/-
+Copyright (c) 2024 Sophie Morel. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Sophie Morel
+-/
+import Mathlib.LinearAlgebra.PiTensorProduct
+import Mathlib.Algebra.DirectSum.Module
+import Mathlib.LinearAlgebra.Multilinear.DirectSum
+
+/-!
+# Tensor products of direct sums
+
+This file shows that taking `PiTensorProduct`s commutes with taking `DirectSum`s in all arguments.
+
+## Main results
+
+* `PiTensorProduct.directSum`: the linear equivalence between a `PiTensorProduct` of `DirectSum`s
+and the `DirectSum` of the `PiTensorProduct`s.
+-/
+
+suppress_compilation
+
+namespace PiTensorProduct
+
+open PiTensorProduct DirectSum LinearMap
+
+open scoped TensorProduct
+
+variable (R : Type*) [CommSemiring R]
+
+variable {ι : Type*} [Fintype ι] [DecidableEq ι]
+
+variable {κ : ι → Type*} [(i : ι) → DecidableEq (κ i)]
+
+variable (M : (i : ι) → κ i → Type*) (M' : Type*)
+
+variable [Π i (j : κ i), AddCommMonoid (M i j)] [AddCommMonoid M']
+
+variable [Π i (j : κ i), Module R (M i j)] [Module R M']
+
+variable [Π i (j : κ i) (x : M i j),  Decidable (x ≠ 0)]
```

Comment:
Here too

Reply:
Ok. I think I found all extra blank lines between variables (I also fixed the proof that wasn;'t working anymore).